src/Application/GamePathResolver.cpp

```diff
@@ -0,0 +1,89 @@
+#include <vector>
+
+#include "Application/GamePathResolver.h"
+
+#include "Engine/IocContainer.h"
+#include "Library/Logger/Logger.h"
+#include "Platform/Platform.h"
+
+
+using EngineIoc = Engine_::IocContainer;
+
+
+static std::string _ResolvePath(Platform *platform, const char *envVarOverride, const std::vector<const wchar_t *> &registryKeys);
+
+
+std::string Application::ResolveMm6Path(Platform *platform) {
+    return _ResolvePath(
+        platform,
+        mm6PathOverrideKey,
+        {
+            L"HKEY_LOCAL_MACHINE/SOFTWARE/GOG.com/Games/1207661253/PATH",
+            L"HKEY_LOCAL_MACHINE/SOFTWARE/GOG.com/GOGMM6/PATH",
+            L"HKEY_LOCAL_MACHINE/SOFTWARE/New World Computing/Might and Magic\x00AE VI/1.0/AppPath",
+        }
+    );
+}
+
+
+std::string Application::ResolveMm7Path(Platform *platform) {
+    return _ResolvePath(
+        platform,
+        mm7PathOverrideKey,
+        {
+            L"HKEY_LOCAL_MACHINE/SOFTWARE/GOG.com/Games/1207658916/Path",
+            L"HKEY_LOCAL_MACHINE/SOFTWARE/GOG.com/GOGMM7/PATH",
+            L"HKEY_LOCAL_MACHINE/SOFTWARE/New World Computing/Might and Magic VII/1.0/AppPath",
+        }
+    );
+}
+
+
+std::string Application::ResolveMm8Path(Platform *platform) {
+    return _ResolvePath(
+        platform,
+        mm8PathOverrideKey,
+        {
+            L"HKEY_LOCAL_MACHINE/SOFTWARE/GOG.com/GOGMM8/PATH",
+            L"HKEY_LOCAL_MACHINE/SOFTWARE/New World Computing/Might and Magic Day of the Destroyer/1.0/AppPath",
+        }
+    );
+}
+
+
+static std::string _ResolvePath(
+    Platform *platform,
+    const char *envVarOverride,
+    const std::vector<const wchar_t *> &registryKeys
+) {
+#ifdef __ANDROID__
+    // TODO: find a better way to deal with paths and remove this android specific block.
+    result = platform->StoragePath(ANDROID_STORAGE_EXTERNAL);
+    if (result.empty()) {
+        result = platform->StoragePath(ANDROID_STORAGE_INTERNAL);
+    }
+
+    return result;
+#else
+    const char *envPathStr = std::getenv(envVarOverride);
+
+    std::string envPath{};
+    if (envPathStr) {
+        envPath = envPathStr;
+    }
+
+    if (!envPath.empty()) {
+        EngineIoc::ResolveLogger()->Info("Path override provided: %s=%s", envVarOverride, envPathStr);
+        return envPath;
+    }
+
+    for (auto key : registryKeys) {
+        envPath = platform->WinQueryRegistry(key);
+        if (!envPath.empty()) {
+            return envPath;
+        }
+    }
+
+    return ".";
```

Comment:
Is there a reason why you returning "." now? Previously it was expected that you return empty string if path is not found. No android is returning empty string and all other platforms ".". And .empty() checks down the road are not making any sense now, not to mentioning you broke message which now says that user should copy files to "."...